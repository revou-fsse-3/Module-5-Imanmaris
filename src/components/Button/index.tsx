import { useMemo } from "react";

interface Props {
    label: string;
    className?: string;
    onClick: () => void;
    variant?: 'primary' | 'second';

}


const Button = ({label, className, onClick, variant='primary'}: Props) => {

    const style = useMemo(
        () => {
            if(variant === 'primary') {
                return "text-white bg-blue-700 hover:bg-blue-800 focus:outline-none focus:ring-4 focus:ring-blue-300 font-medium rounded-full text-sm px-5 py-2.5 text-center me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800";
            }

            return "py-2.5 px-5 me-2 mb-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-full border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700";
            // "hover:bg-indigo-950 font-bold text-white bg-orange-400 rounded-full px-5 py-3"
        },
        []
    )

    return (
        <>
            <button className={`${className} ${style}`}
            onClick={onClick}>
                {label}
            </button>
        </>
    )
}

export default Button;