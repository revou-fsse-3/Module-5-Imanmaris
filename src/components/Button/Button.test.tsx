
import {fireEvent, render, screen, waitFor} from '@testing-library/react';
import Button from '.';

describe ('Unit Test for Button Component', () => {

    const mocking = {
        onSubmit : jest.fn()   
    }

    test('Button render correctly', () => {
        
        // const document = render(<Button label={'Button testing'} onClick={() => console.log('asd')}/>);
        const document = render(<Button label={'Button testing'} onClick={mocking.onSubmit}/>);
        expect(document).toMatchSnapshot();
    })

    test('Button should be able to click', async () => {
        
        render(<Button label={'Button testing'} onClick={mocking.onSubmit}/>);
        const button = screen.getByText('Button testing');
        fireEvent.click(button);

        await waitFor ( () => {
            expect(mocking.onSubmit).toHaveBeenCalledTimes(1);
        })
    })

});