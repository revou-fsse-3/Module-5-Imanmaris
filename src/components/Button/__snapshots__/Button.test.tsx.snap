// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test for Button Component Button render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="undefined text-white bg-blue-700 hover:bg-blue-800 focus:outline-none focus:ring-4 focus:ring-blue-300 font-medium rounded-full text-sm px-5 py-2.5 text-center me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
      >
        Button testing
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="undefined text-white bg-blue-700 hover:bg-blue-800 focus:outline-none focus:ring-4 focus:ring-blue-300 font-medium rounded-full text-sm px-5 py-2.5 text-center me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
    >
      Button testing
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
